whylog/config/__init__.py

```diff
@@ -103,6 +114,46 @@ def _filter_rule_set(self, parsers_list):
     def _get_locations_for_logs(self, logs_types_list):
         pass
 
+    def is_free_parser_name(self, parser_name):
+        return parser_name not in self._parsers
+
+    def _is_unique_parser_name(self, parser_name, black_list):
+        return self.is_free_parser_name(parser_name) and (parser_name not in black_list)
+
+    def propose_parser_name(self, line, regex_str, black_list):
+        building_words = AbstractConfig._get_building_words(line, regex_str)
+        if not building_words:
+            return uuid.uuid4()
+        if len(building_words) == 1:
+            if self._is_unique_parser_name(building_words[0], black_list):
+                return building_words[0]
+            else:
+                return self._find_free_by_number_appending(building_words[0], black_list)
+        for i in range(len(building_words) - 1):
```

Comment:
I would rather remove this method of generating names.
For line: "2016-04-12 23:54:45 Connection error occurred on comp1. Host name: host1"
You can have names like: 'connection_error', 'error_occurred', 'occurred_on'
It will be is easier for user to find parser by prefix rather than random word inside parser. I suppose it alse will be easier to remember. We will have better order in Whylog database while manually searching/scrolling through it.